proxy/backendtimeout_test.go

```diff
@@ -0,0 +1,200 @@
+package proxy
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+)
+
+func TestSlowService(t *testing.T) {
+	service := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(2 * time.Millisecond)
+	}))
+	defer service.Close()
+
+	doc := fmt.Sprintf(`* -> backendTimeout("1ms") -> "%s"`, service.URL)
```

Comment:
this might always be ok in CI tests

Reply:
one way to test this better could be a service that never responds:

```
<-testDone
```